[PL-46853] Correct version compatibility, clarify text (#5271)
harness/developer-hub: docs/platform/delegates/install-delegates/delegate-upgrades-and-expiration.md

## Delegate expiration policy

> Delegates expire six months (24 weeks) from the installation date. However, the expiration of delegates does not mean that they stop working altogether. Delegates are backward-compatible. However, there is a possibility of facing issues if the backend has moved too far ahead, and the delegate is no longer forward-compatible.

Author: ```suggestion
Delegates expire six months (24 weeks) from the installation date. Delegates do not stop working after expiration. They are backward-compatible but may face issues if the backend has moved too far ahead and the delegate is no longer forward-compatible.
```
